packages/webapp/package.json

```diff
@@ -28,8 +28,8 @@
     "next": "^12.2.2",
     "next-pwa": "^5.5.4",
     "next-seo": "^5.4.0",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
```

Comment:
`next@12.x` is not compatible with React 18 I think, probably works because react is backwards compatible but due to core `react-dom` changes it might break in some weird cases (as you had below), we should maybe leave on `17.x` for webapp until Next upgrade?

Reply:
Interesting. Yeah, we could maybe have avoided some of the weird things introduced here if we did that. I will try it tomorrow and see how it turns out.

Reply:
After testing to install react-dom v17.x and react on 18.x - there seem to be some package issues after clearing up all the modules and then run the local env.

I think we should be safe having updated react-dom. Like you mentioned, React 18 is backwards compatible, and on the NextJS side, in order to use the React 18 features, we will have to add some experimental flags within the config. That means the default setting is without the React 18 features integrated.

<img width="556" alt="Screenshot 2024-07-02 at 2 12 59 PM" src="https://github.com/dailydotdev/apps/assets/13744167/c7865a66-a55f-4fe4-8861-d8657ae1d8be">

Reply:
I checked the package.json for next.js v12.3.4, and they used react 18. But they also had tests for react 17 for backwards compatibility, so I think we're safe to upgrade to v18.

Reply:
I know 13.x and above has the internal check to only use 18 features when in app router, but was not sure if they backported to 12.x. Luckly we don't plan on staying on 12.x for long so if we don't find anything in testing its good to go. Tnx for checking!